src/main/java/journeymap/client/render/map/GridRenderer.java

```diff
@@ -249,9 +253,10 @@ public void updateTiles(MapType mapType, int zoom, boolean highQuality, int widt
         // Derive offsets for centering the map
         Point2D blockPixelOffset = centerTile.blockPixelOffsetInTile(centerBlockX, centerBlockZ);
         final double blockSizeOffset = Math.pow(2, zoom) / 2;
-        final int magic = (gridSize == 5 ? 2 : 1) * Tile.TILESIZE; // TODO:  Understand why "2" as it relates to gridSize.  If gridSize is 3, this has to be "1".
 
-        double displayOffsetX = xOffset + magic - ((srcSize - lastWidth) / 2);
+        // TODO: Why is the extra offset necessary and what are we subtracting here? (2x)
+        int extraOffsetX = (gridSizeWidth / 2) * Tile.TILESIZE;
```

Comment:
this extra offset is the (tiles from center * tilesize), I dont exactly remember why this was needed, this was removed with the grid renderer(now maprenderer) rewrite in 6.0.0.